dbms/src/Dictionaries/HashedDictionary.cpp

```diff
@@ -446,7 +446,7 @@ void HashedDictionary::addAttributeSize(const Attribute & attribute)
         /** TODO: more accurate calculation */
         bytes_allocated += sizeof(CollectionType<T>);
         bytes_allocated += bucket_count;
-        bytes_allocated += map_ref->size() * sizeof(Key) * sizeof(T);
+        bytes_allocated += map_ref->size() * (sizeof(Key) + sizeof(T));
```

Comment:
There was a real bug? What should work better now?

Reply:
Yes, looks like the amount of allocated space was not calculated correcly in system.dictionaries for sparse_hashed dictionary layout. We may ask @azat to make sure.

Reply:
Yep, not it looks sane

FWIW `SparseCollectionType` should be used over `CollectionType` here (fortunately this does not affects the result a lot)

P.S. does `clang-tidy` automatically finds this? Interesting!

Reply:
> Yep, not it looks sane

You mean "now it looks sane".

> FWIW SparseCollectionType should be used over CollectionType here

Ok, will fix.

> does clang-tidy automatically finds this? Interesting!

Yes, it is https://clang.llvm.org/extra/clang-tidy/checks/bugprone-sizeof-expression.html

Reply:
>You mean "now it looks sane".

Yep!